### ISSUE TITLE
Synchronization with Outlook does not work

### ISSUE THREAD

**Issue by a1ex-ak** [nextcloud/mail#10193]
### Steps to reproduce

1. Two-factor authentication enabled
2. Application password created
3. Incorrect IMAP login or password

### Expected behavior

Until yesterday 25.09.2024 the synchronization worked

### Actual behavior

Incorrect IMAP login or password

### Mail app version

3.7.8

### Mailserver or service

Outlook

### Operating system

Debian 12

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database

MariaDB

**Comment by jaycedk** [comment]
Seeing the samething.
Not sure if has something to with this where basic authentication is no longer used.
https://support.microsoft.com/en-us/office/modern-authentication-methods-now-needed-to-continue-syncing-outlook-email-in-non-microsoft-email-apps-c5d65390-9676-4763-b41f-d7986499a90d

**Comment by a1ex-ak** [comment]
> Видя заметку. Не уверен, есть ли что-то с этим, где базовая аутентификация больше не используется. https://support.microsoft.com/en-us/office/modern-authentication-methods-now-needed-to-continue-syncing-outlook-email-in-non-microsoft-email-apps-c5d65390-9676-4763-b41f-d7986499a90d

So what to do now? ))

**Comment by jaycedk** [comment]
Us as users can't do much, but hope Nextcloud mail app will conform to use this new method of authentication.

**Comment by ChristophWurst** [comment]
Do you mean Exchange? This app support OAuth for Microsoft. Use that instead of (app) passwords

**Comment by jaycedk** [comment]
Well.. In my case its a free hotmail account.
sadhg@hotmail.com
![image](https://github.com/user-attachments/assets/02e05fa1-3d7a-424a-b80d-cd168f91d3de)
If I switch to Manuel I don't OAuth in the selection. 
![image](https://github.com/user-attachments/assets/43b66ba6-0b52-4be7-b593-a5e4ecb47d86)
It might be me doing something very wrong 🤣😂

**Comment by jaycedk** [comment]
Well how do I get to that.
Its a hotmail account ( free account ) without access to a Azure AD.
I can't make these changes to my account.

**Comment by ChristophWurst** [comment]
You're right. I don't know how these accounts can be connected otherwise. According to https://answers.microsoft.com/en-us/outlook_com/forum/all/are-app-passwords-being-discontinued/0030efe5-87fa-48d5-b14c-16f355ebb5fa the change came September 16th.

**Comment by jaycedk** [comment]
Yes I linked the changes above
https://support.microsoft.com/en-us/office/modern-authentication-methods-now-needed-to-continue-syncing-outlook-email-in-non-microsoft-email-apps-c5d65390-9676-4763-b41f-d7986499a90d

**Comment by Cathnan** [comment]
Hi, I didn't notice this change either and ran into a non-working personal hotmail account in my nextcloud mail app. I've been tinkering around a while until I found this page. The tutorial posted by ChristophWurst worked perfectly for me. Thank you. Could it be mentioned either in the mail app or the groupware page in the settings? @jaycedk you may want to check out [portal.azure.com](https://azure.microsoft.com/en-gb/pricing/purchase-options/azure-account/) I could sign up for a free pay as you go account, which seems like all you need to follow the tutorial. But I'm not sure because I also linked my hotmail account to microsoft entra while tinkering.

**Comment by mat-m** [comment]
> You're right. I don't know how these accounts can be connected otherwise. According to https://answers.microsoft.com/en-us/outlook_com/forum/all/are-app-passwords-being-discontinued/0030efe5-87fa-48d5-b14c-16f355ebb5fa the change came September 16th.

That's strange because registration worked for K-9 yesterday with the same settings. I was in plain password auth and I was also blocked. Now, my settings in K-9 are:
![Image](https://github.com/user-attachments/assets/7f230801-0d69-40b1-b3fd-2fc5b92190c6)

There is "just" a setting for authentication between plain, encrypted password, client cert and OAuth2., BUt I had nothing to change in my Outlook.com account to make it work. Maybe the checkbox "Send client information" is also needed.

https://account.microsoft.com/privacy/app-access
shows:
![Image](https://github.com/user-attachments/assets/b2756e72-7916-4c2e-add2-1f8af78d7522)

For the record, my setting in NC:
![Image](https://github.com/user-attachments/assets/540cde2c-676e-4326-a947-84c66135357c)

**Comment by P1zz4br0etch3n** [comment]
Using a personal/non-working live.de account, I was able to add a Microsoft Entra ID App Registration on portal.azure.com after hitting "ignore" in a warning dialog a couple times. I didn't need to start a free plan/trial to do so. But after successfully setting it up in the Nextcloud Groupware section I'm still not able to add my account to the mails app. It just says there was an error setting up the account. In the dev tools console I found this error message (I replaced uid and domain for privacy reasons):

```
[ERROR] mail: could not save account details 
Object { app: "mail", uid: "user", level: 0, error: {…} }
​
app: "mail"
​
error: Object { stack: "K@https://cloud.domain.com/custom_apps/mail/js/mail.js?v=908bc780-5:2:986102\nje@https://cloud.domain.com/custom_apps/mail/js/mail.js?v=908bc780-5:2:997518\nf@https://cloud.domain.com/custom_apps/mail/js/mail.js?v=908bc780-5:2:1002159\n", message: "Request failed with status code 500", name: "AxiosError", … }
​​
code: "ERR_BAD_RESPONSE"
​​
config: Object { timeout: 0, xsrfCookieName: "XSRF-TOKEN", xsrfHeaderName: "X-XSRF-TOKEN", … }
​​
message: "Request failed with status code 500"
​​
name: "AxiosError"
​​
request: XMLHttpRequest { readyState: 4, timeout: 0, withCredentials: false, … }
​​
response: Object { data: {…}, status: 500, statusText: "", … }
​​
stack: "K@https://cloud.domain.com/custom_apps/mail/js/mail.js?v=908bc780-5:2:986102\nje@https://cloud.domain.com/custom_apps/mail/js/mail.js?v=908bc780-5:2:997518\nf@https://cloud.domain.com/custom_apps/mail/js/mail.js?v=908bc780-5:2:1002159\n"
​​
<prototype>: Object { constructor: K(e, t, n, r, a), toJSON: toJSON(), stack: "", … }
​
level: 0
​
uid: "user"
​
<prototype>: Object { … }
<anonymous code>:1:145535
    overrideMethod <anonymous code>:1
    value ConsoleLogger.js:74
    value ConsoleLogger.js:100
    onSubmit AccountForm.vue:634
```

Also I think it would be much easier for regular Nextcloud users if the mail app itself could implement the oauth2 authentication, just like e.g. the gmail app on android. Or is it impossible due to everybody's using a different domain? 🤔

**Comment by mat-m** [comment]
Maybe application passwords are working, but you need to have 2FA authentication enabled.

**Comment by ChristophWurst** [comment]
> [@ChristophWurst](https://github.com/ChristophWurst) :what can I do to check what's wrong ?
> 
> Message from Mail 4.0.2 on NC30 is `IMAP username or password is wrong`

You could try with https://github.com/nextcloud/mail/blob/main/doc/admin.md#logging and inspect the IMAP communication. There might be more info there.

**Comment by mat-m** [comment]
Hello

One output through occ:

```
# /var/www/html/occ mail:account:sync  -vvv 7
[debug] Getting namespaces for account 7 failed: Mail server denied authentication.

In MailboxSync.php line 109:

  [OCA\Mail\Exception\ServiceException (102)]
  IMAP error synchronizing account 7: Mail server denied authentication.


Exception trace:
  at /var/www/html/custom_apps/mail/lib/IMAP/MailboxSync.php:109
 OCA\Mail\IMAP\MailboxSync->sync() at /var/www/html/custom_apps/mail/lib/Command/SyncAccount.php:87
 OCA\Mail\Command\SyncAccount->sync() at /var/www/html/custom_apps/mail/lib/Command/SyncAccount.php:72
 OCA\Mail\Command\SyncAccount->execute() at /var/www/html/3rdparty/symfony/console/Command/Command.php:298
 Symfony\Component\Console\Command\Command->run() at /var/www/html/3rdparty/symfony/console/Application.php:1040
 Symfony\Component\Console\Application->doRunCommand() at /var/www/html/3rdparty/symfony/console/Application.php:301
 Symfony\Component\Console\Application->doRun() at /var/www/html/3rdparty/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at /var/www/html/lib/private/Console/Application.php:183
 OC\Console\Application->run() at /var/www/html/console.php:87
 require_once() at /var/www/html/occ:11

In Socket.php line 572:

  [Horde_Imap_Client_Exception (102)]
  Mail server denied authentication.


Exception trace:
  at /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Socket.php:572
 Horde_Imap_Client_Socket->_login() at /var/www/html/custom_apps/mail/lib/IMAP/HordeImapClient.php:59
 OCA\Mail\IMAP\HordeImapClient->_login() at /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php:853
 Horde_Imap_Client_Base->login() at /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php:1404
 Horde_Imap_Client_Base->listMailboxes() at /var/www/html/custom_apps/mail/lib/IMAP/FolderMapper.php:52
 OCA\Mail\IMAP\FolderMapper->getFolders() at /var/www/html/custom_apps/mail/lib/IMAP/MailboxSync.php:106
 OCA\Mail\IMAP\MailboxSync->sync() at /var/www/html/custom_apps/mail/lib/Command/SyncAccount.php:87
 OCA\Mail\Command\SyncAccount->sync() at /var/www/html/custom_apps/mail/lib/Command/SyncAccount.php:72
 OCA\Mail\Command\SyncAccount->execute() at /var/www/html/3rdparty/symfony/console/Command/Command.php:298
 Symfony\Component\Console\Command\Command->run() at /var/www/html/3rdparty/symfony/console/Application.php:1040
 Symfony\Component\Console\Application->doRunCommand() at /var/www/html/3rdparty/symfony/console/Application.php:301
 Symfony\Component\Console\Application->doRun() at /var/www/html/3rdparty/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at /var/www/html/lib/private/Console/Application.php:183
 OC\Console\Application->run() at /var/www/html/console.php:87
 require_once() at /var/www/html/occ:11

mail:account:sync [-f|--force] [--] <account-id>

```

I also got this in logging:

```
[no app in context] Debug: dirty table reads: SELECT * FROM `*PREFIX*mail_mailboxes` WHERE (`account_id` = :dcValue1) AND (`name_hash` = :dcValue2)
	POST /apps/mail/api/mailboxes/92/sync
	from 192.168.1.254 by mm at 17 nov. 2024, 20:52:21

[mail] Error: Error when communicating with the mail server.
	POST /apps/mail/api/mailboxes/150/sync
	from 192.168.1.254 by mm at 17 nov. 2024, 20:54:47

```

Since the account is blocked, I can't check through UI if mailboxes 92 and 150 belong to account 7.

**Comment by mat-m** [comment]
It seems that connecting with pre-generated application password is not working. GMail imports is also failing.

Thunderbird is creating its own app password by triggering a webpage to connect to MS servers. IIRC, K9-Mail did the same thing: they authenticate with OAuth2 protocol which does that.

Although MS allows to create app passwords beforehand, authentication protocol seems to be different from what is implemented for NC Mail or Gmail.

**Comment by mat-m** [comment]
mutt has oauth2 support through additional  [script](https://gitlab.com/muttmua/mutt/-/blob/master/contrib/mutt_oauth2.py) (through https://github.com/ork/mutt-office365/issues/28)
I don't know if there something broken in MS app passwords or if they actually only support OAuth2 (like gmail ?)

**Comment by ChristophWurst** [comment]
Dear @mat-m, thanks for looking into this. To my knowledge Microsoft has indeed stopped any kind of basic authentication. Therefore even app passwords no longer work. See https://answers.microsoft.com/en-us/outlook_com/forum/all/are-app-passwords-being-discontinued/0030efe5-87fa-48d5-b14c-16f355ebb5fa.

This app can authenticate Microsoft and Google accounts via OAuth. However, we don't ship any client IDs or client secrets, we leave this to the admins of the installation. See https://docs.nextcloud.com/server/latest/admin_manual/groupware/mail.html#xoauth2-authentication-with-microsoft-azure-ad.

**Comment by mat-m** [comment]
@ChristophWurst: Bad thing is that free accounts like @jaycedk or me (*@outlook.com, *@hotmail.com/fr/...) do not have access to the Azure portal. So I guess wed're dead on this one.

**Comment by ChristophWurst** [comment]
Thanks for the feedback, @mat-m. Do you have any suggestions for a workaround or solution?

**Comment by mat-m** [comment]
I guess we have none unless having our own OAuth client. I don't know if K-9 mail is also doing this.

**Comment by MarvinMynx** [comment]
I' having trouble registreing the Azure App:

https://docs.nextcloud.com/server/latest/admin_manual/groupware/mail.html#account-delegation

> Have a look at step 8 on where to find the redirect URI. Finally, click on register to proceed.

There is no URI nor a Step 8. 🤔

**Comment by SebastianKrupinski** [comment]
Hi @MarvinMynx 

The instruction for the redirection url are in the documentation you linked.

![Image](https://github.com/user-attachments/assets/f343f8cb-b061-44ce-bd4c-058b3a88d4a8)

**Comment by ChristophWurst** [comment]
> [@ChristophWurst](https://github.com/ChristophWurst) , maybe you can reach to [@cketti](https://github.com/cketti) about this.

Good idea, @mat-m! Could you take the lead on this?

**Comment by MarvinMynx** [comment]
> Hi [@MarvinMynx](https://github.com/MarvinMynx)
> 
> The instruction for the redirection url are in the documentation you linked.
> 
> ![Image](https://github.com/user-attachments/assets/f343f8cb-b061-44ce-bd4c-058b3a88d4a8)

Oh, I missed this part—sorry! I got this part working, but I'm using a reverse proxy for Nextcloud to enable internet access, as well as CardDAV and CalDAV access. However, for the OAuth request, I get the following error:

> invalid_request: The provided value for the input parameter 'redirect_uri' is not valid. The expected value is a URI that matches a redirect URI registered for this client application.

I don't see any documentation or comments on how to configure redirect schemes on a reverse proxy. Can someone help me with this?

**Comment by MattRGX** [comment]
Any news on that issue ? as today, i still can't connect my hotmail account to the mail app.

**Comment by ChristophWurst** [comment]
Thank you for your message @MattRGX! We have triaged and accepted the feature request at https://github.com/nextcloud/mail/issues/10193#event-14454897198. It's open to be picked by someone. Hope this makes the status clearer.

**Comment by mat-m** [comment]
> > [@ChristophWurst](https://github.com/ChristophWurst) , maybe you can reach to [@cketti](https://github.com/cketti) about this.
> 
> Good idea, [@mat-m](https://github.com/mat-m)! Could you take the lead on this?

Got on answers so far.

**Comment by addohm** [comment]
> Thank you for your message [@MattRGX](https://github.com/MattRGX)! We have triaged and accepted the feature request at [#10193 (comment)](https://github.com/nextcloud/mail/issues/10193#event-14454897198). It's open to be picked by someone. Hope this makes the status clearer.

I appreciate your attention on this.  Aside from my own private emails, I also have several business accounts between my wife and I that need to migrate over to Nextcloud that cant.

**Comment by JMarcosHP** [comment]
Well, it's time to fallback to thunderbird and disable mail app until they implement an OAuth client.

**Comment by ChristophWurst** [comment]
No, that's the problem. I don't think there is a solution for self-hosted webmail.

**Comment by JMarcosHP** [comment]
At least provide an alternative for selfhosted users, and leave the current configuration for organizations/enterprises who want to use Entra ID.

**Comment by ChristophWurst** [comment]
Yes, I'd really like to, but how? Our app did not change in this regard. It's Microsoft that stopped app password authentication, forcing people to OAuth, which is tricky for self hosting.

**Comment by TryThisOneMyFriend** [comment]
> Yes, I'd really like to, but how? Our app did not change in this regard. It's Microsoft that stopped app password authentication, forcing people to OAuth, which is tricky for self hosting.

I'm using sendmail. It is working but the emails will not be verified by Outlook or Gmail, so they will initially go to spam. If you are selfhosting that shouldn't be a problem

**Comment by mat-m** [comment]
XThunderbird for Android (TfA) uses OAuth 2.0, without app password. It has required access to a service and registered itself into my google account, after requesting double-step vérification. Authorization can be checked here: https://myaccount.google.com/connections/overview, then selecting Thunderbird.

IFTTT (ifttt.com), only used through the website, can connect to a Gmail or Outlook account without creating any app password, nor registering any service in Entra. 
I guess they manage a private token to authenticate on the service once we validated the authorization.



I don't know if there is a php OAuth2 Library to do the same, but worth checking. 
[zorn-v/nextcloud-social-login](https://deepwiki.com/zorn-v/nextcloud-social-login/3.1-oauth2-providers) seems to support Google by default. 


For Microsoft, URL looks like https://account.live.com/Consent/Update?mkt=FR-FR&uiflavor=web&id=xxxx&ru=https://login.live.com/oauth20_authorize.srf%

**Comment by AhmedViper** [comment]
>There is no way of "getting around". This is how OAuth works.


Can I suggest adding a 'supported/tested' email providers somewhere in the documentation / home which clarifies supported or at least known unsupported mail providers ?

As a new user It was hard getting from 'Incorrect IMAP login or password' (which is very misleading, I kept trying my password a different times thinking I put it wrong) to this issue then to Outlook being *basically* not supported anymore.

**Comment by ChristophWurst** [comment]
Sorry, but that is not true. You have to set up oauth but then MS365 works just fine.
I understand that this is inconvenient and that desktop application have an advantage here.

**Comment by ChristophWurst** [comment]
> As a new user It was hard getting from 'Incorrect IMAP login or password' (which is very misleading, I kept trying my password a different times thinking I put it wrong)

I've tracked this in https://github.com/nextcloud/mail/issues/12779.

**Comment by AhmedViper** [comment]
> Can I suggest adding a 'supported/tested' email providers somewhere in the documentation / home which clarifies supported or at least known unsupported mail providers ?

> Sorry, but that is not true. You have to set up oauth but then MS365 works just fine.

I misspoke, what I meant by unsupported is not 'out of the box' ready to use, basically making it clearer that in order to use X email provider you have to have/setup/do Y *and potentially, how to do it or a link to a guide maybe?*
